Mathlib/Data/List/Basic.lean

```diff
@@ -247,12 +247,17 @@ theorem singleton_eq (x : α) : ({x} : List α) = [x] :=
   rfl
 #align list.singleton_eq List.singleton_eq
 
-theorem insert_neg [DecidableEq α] {x : α} {l : List α} (h : x ∉ l) : Insert.insert x l = x :: l :=
-  if_neg h
+theorem insert_neg [DecidableEq α] {x : α} {l : List α} (h : x ∉ l) :
+    Insert.insert x l = x :: l := by
+  dsimp [Insert.insert, List.insert]
+  rw [if_neg]
+  simp_all only [elem_iff, not_false_eq_true, h]
```

Comment:
Should this be proven with (or deprecated in favor of) `insert_of_not_mem`?

Reply:
Done.